### PR TITLE
Support error bounds in the form of floats

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR allows error bounds for `FIELD`s to be specified as `f32` instead of `u64`.

### CODE REVIEW

**Review by CGodiksen — APPROVED** [review]

File: crates/modelardb_server/src/parser.rs [hunk 1]
```diff
@@ -176,6 +176,20 @@ impl ModelarDbDialect {
         }
     }
 
+    /// Return its value as a [`f32`] if the next [`Token`] is a
+    /// [`Token::Number`], otherwise a [`ParserError`] is returned.
+    fn parse_positive_literal_f32(&self, parser: &mut Parser) -> Result<f32, ParserError> {
+        let token_with_location = parser.next_token();
+        match token_with_location.token {
+            Token::Number(maybe_f32, _) => maybe_f32.parse::<f32>().map_err(|error| {
```
Comment (chrthomsen) [inline crates/modelardb_server/src/parser.rs:184]: Is this locale-dependent, i.e., 0.5 for en_US, but 0,5 for da_DK?

File: crates/modelardb_server/src/parser.rs [hunk 2]
```diff
@@ -681,7 +695,7 @@ mod tests {
 
     #[test]
     fn test_tokenize_and_parse_create_model_table() {
-        let sql = "CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field_one FIELD, field_two FIELD(10), tag TAG)";
+        let sql = "CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field_one FIELD, field_two FIELD(10.5), tag TAG)";
```
Comment (chrthomsen) [inline crates/modelardb_server/src/parser.rs:698]: Can't this fail for locale da_DK if float parsing is locale-dependent?

File: crates/modelardb_server/src/parser.rs (hunk 1 above)
Comment (skejserjensen) [reply crates/modelardb_server/src/parser.rs:184]: Based on the documentation it seems only `en_US` is supported as `f32`s can only be parsed if they follow this [grammar](https://doc.rust-lang.org/stable/std/primitive.f32.html#method.from_str).

File: crates/modelardb_server/src/parser.rs (hunk 2 above)
Comment (skejserjensen) [reply crates/modelardb_server/src/parser.rs:698]: Based on the documentation it seems only `en_US` is supported as `f32`s can only be parsed if they follow this [grammar](https://doc.rust-lang.org/stable/std/primitive.f32.html#method.from_str).

Comment (skejserjensen) [reply crates/modelardb_server/src/parser.rs:698]: Based on the documentation it seems only `en_US` is supported as `f32`s can only be parsed if they follow this [grammar](https://doc.rust-lang.org/stable/std/primitive.f32.html#method.from_str).

File: crates/modelardb_server/src/parser.rs (hunk 1 above)
Comment (chrthomsen) [reply crates/modelardb_server/src/parser.rs:184]: OK, good.

File: crates/modelardb_server/src/parser.rs (hunk 2 above)
Comment (chrthomsen) [reply crates/modelardb_server/src/parser.rs:698]: Good, this is not a problem then

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**